river/command/map.zig

```diff
@@ -316,32 +323,58 @@ const OptionalArgsContainer = struct {
     i: usize,
     release: bool,
     repeat: bool,
+    layout_index: ?u32,
 };
 
 /// Parses optional args (such as -release) and return the index of the first argument that is
 /// not an optional argument
 /// Returns an OptionalArgsContainer with the settings set according to the args
 /// Errors cant occur because it returns as soon as it gets an unknown argument
-fn parseOptionalArgs(args: []const []const u8) OptionalArgsContainer {
+fn parseOptionalArgs(args: []const []const u8, seat: *Seat) !OptionalArgsContainer {
     // Set to defaults
     var parsed = OptionalArgsContainer{
         // i is the number of arguments consumed
         .i = 0,
         .release = false,
         .repeat = false,
+        .layout_index = null,
     };
 
-    var i: usize = 0;
-    for (args) |arg| {
-        if (mem.eql(u8, arg, "-release")) {
+    var j: usize = 0;
+    while (j < args.len) : (j += 1) {
+        if (mem.eql(u8, args[j], "-release")) {
             parsed.release = true;
-            i += 1;
-        } else if (mem.eql(u8, arg, "-repeat")) {
+            parsed.i += 1;
+        } else if (mem.eql(u8, args[j], "-repeat")) {
             parsed.repeat = true;
-            i += 1;
+            parsed.i += 1;
+        } else if (mem.eql(u8, args[j], "-layout")) {
+            // TODO: Figure out how we want the interface to look like.
+            // Currently, this uses the full layout name e.g. "English (Colemak)",
+            // because this is what `xkb_keymap_layout_get_index()` provides.
+            // But maybe the short name e.g. "us(colemak)" would be better because
+            // it's already used for XKB_DEFAULT_LAYOUT. We could also parse both.
```

Comment:
I also like the short names better. I've just looked in how to do this. According to this https://xkbcommon.org/doc/current/xkbcommon_8h.html#ad37512642806c55955e1cd5a30efcc39 , the short names are not present in the compiled keymap. So I think we would need to create a temporary keymap from the argument and use `xkb_keymap_layout_get_name(0)` on this to get the full name 

Would that overhead be acceptable to you or do you maybe have a better idea?

Reply:
> Hmm, I guess a third option would be to use layout indexes directly here.

Isn't that's how it was implemented at some point? If this is the simplest option, then why not? I don't think there is a reason to define a mapping for a layout that is not defined in `XKB_DEFAULT_LAYOUT`.

Reply:
> Isn't that's how it was implemented at some point? If this is the simplest option, then why not? I don't think there is a reason to define a mapping for a layout that is not defined in `XKB_DEFAULT_LAYOUT`.

Yeah I had this before I implemented parsing the layout name.

IMO, the advantage of a name is that the user can change the environment variable (e.g. swap order of layouts) without the need to touch the river init.

Reply:
It would still not be possible to use a layout that's not in XKB_DEFAULT_LAYOUT.

Reply:
> It would still not be possible to use a layout that's not in XKB_DEFAULT_LAYOUT.

I think this is actually a pretty strong argument in favor of sticking to layout indexes here as it makes it clear that this is selecting from the list of layouts defined by `XKB_DEFAULT_LAYOUT`. Accepting a name does make it feel like any layout name should work and I don't think we want to support that.

Reply:
Alright then, I reverse this :) Saves us a lot of parsing code :)